src/witan/cic/episodes/transformation_report.clj

```diff
@@ -13,52 +13,90 @@
     :remove  "for removal"
     :edited  "edited"
     :updated "updated"
-    cmd))
+    :examine "examine"
+    (str cmd)))
+
+(def header
+  ["ID" "dob" "sex"
+   "report year" "period id"
+   ;; "reason for new episode" "report date" "ceased" "placement" "legal status" "care status"
+   "RNE" "DECOM" "DEC" "PLACE" "LS" "CIN"
+   "file name" "sheet name" "row number"
+   "edit status" "reason"
+   "affecting report year"
+   ;; "new report date" "new ceased" "new placement" "new legal status" "new cares status"
+   "DECOM" "DEC" "PLACE" "LS" "CIN"
+   "affecting file name" "affecting sheet name" "affecting row number"
+   "original fields"
+   "DECOM" "DEC" "PLACE" "LS" "CIN"])
+
+(defn format-row [episode]
+  (let [fields ((juxt ::wce/id (fn [e] (str (::wce/dob e))) ::wce/sex
+                      ::wce/report-year ::wce/period-id
+                      (fn [e] (str (::wce/reason-new-episode e))) (fn [e] (str (::wce/report-date e))) (fn [e] (str (::wce/ceased e))) ::wce/placement ::wce/legal-status ::wce/care-status
+                      ::wcie/file-name ::wcie/sheet-name ::wcie/row-index)
+                episode)
+        edit (or
+              (some #(when (= (::wce/command %) :remove) %) (::wce/edit episode))
+              (first (::wce/edit episode)))]
+    (cond
+      (::wce/new edit)
+      (let [new (::wce/new edit)
+            previous (::wce/previous edit)]
+        (into fields
+              [(user-friendly-command-string (::wce/command edit)) (::wce/reason edit)
+               (::wce/report-year new)
+               (str (::wce/report-date new)) (str (::wce/ceased new)) (::wce/placement new) (::wce/legal-status new) (::wce/care-status new)
+               (::wcie/file-name new) (::wcie/sheet-name new) (::wcie/row-index new)
+               "Original" (str (::wce/report-date previous)) (str (::wce/ceased previous)) (::wce/placement previous) (::wce/legal-status previous) (::wce/care-status previous)]))
+
+      (::wce/replacement edit)
+      (let [replacement (::wce/replacement edit)]
+        (into fields [(user-friendly-command-string (::wce/command edit)) (::wce/reason edit)
+                      (::wce/report-year replacement)
+                      (str (::wce/report-date replacement)) (str (::wce/ceased replacement)) (::wce/placement replacement) (::wce/legal-status replacement) (::wce/care-status replacement)
+                      (::wcie/file-name replacement) (::wcie/sheet-name replacement) (::wcie/row-index replacement)]))
```

Comment:
this formatting I find very confusing

Reply:
Is it just sorting and organising?

Reply:
I'll come back to this again and see if I can make it prettier later.